packages/integrations/vercel/src/serverless/adapter.ts

```diff
@@ -522,9 +521,15 @@ function getRuntime(process: NodeJS.Process, logger: AstroIntegrationLogger): Ru
 		);
 		return 'nodejs18.x';
 	}
-	if (support.status === 'current') {
+	if (support.status === 'default') {
 		return `nodejs${major}.x`;
 	}
+	if (support.status === 'retiring' && support.warnDate && new Date() >= support.warnDate) {
+	  logger.warn(
+      `Your project is being built for Node.js ${major} as the runtime, which is retiring by ${support.removal}.`
+    );
+    return `nodejs${major}.x`;
+	}
```

Comment:
Should this be something like:
```suggestion
	if (support.status === 'retiring') {
	  if (support.warnDate && new Date() >= support.warnDate) {
	    logger.warn(
        `Your project is being built for Node.js ${major} as the runtime, which is retiring by ${support.removal}.`
      );
    }
    return `nodejs${major}.x`;
	}
```

Otherwise currently (before the warn date) it won't return the string.